src/partitioning/parmetis_partitioner.C

```diff
@@ -181,7 +181,7 @@ void ParmetisPartitioner::_do_repartition (MeshBase & mesh,
                                        &mpi_comm);
 
   // Assign the returned processor ids
-  this->assign_partitioning (mesh);
+  this->assign_partitioning (mesh, reinterpret_cast<std::vector<dof_id_type> &> (_pmetis->part));
```

Comment:
If Parmetis::idx_t is not the same as dof_id_type, doesn't this break in horrifying ways?

Reply:
So they are guaranteed to be the same size, but may be of different signedness. As long as METIS doesn't hand us back any negative values this should be OK, since `unsigned foo` can fit all non-negative values of `foo` for the integral types.

Reply:
See also contrib/metis/metis.h where we set `IDXTYPEWIDTH` according to the value of `LIBMESH_DOF_ID_BYTES`.

Reply:
We can always fall back on a copy if you want to be paranoid, but I discussed this code with @fdkong and it seemed benign enough to me.

Reply:
And if we snoop ParMETIS from PETSc, then we guarantee that dof_id_type is the size of PetscInt and they presumably guarantee that PetscInt is the same size as the ParMETIS index type, so that'll still work too?

Ok, seems safe.  But could we through a static_cast in here, just to be paranoid?

Reply:
> Ok, seems safe. But could we throw a static_cast in here, just to be paranoid?

Has to be `reinterpret_cast`, I'm afraid. Compilers don't allow `static_cast` between `vector<X>` and `vector<Y>` even if the types `X` and `Y` are cast'able.

Reply:
Sorry, I'm an idiot. I meant to type static_assert, not static_cast.  Like we have in petsc_vector.h for assuring sizeof equality between PetscInt and dof_id_type.